airflow/models/taskinstance.py

```diff
@@ -1144,7 +1144,10 @@ def next_retry_datetime(self):
             delay = timedelta(seconds=delay_backoff_in_seconds)
             if self.task.max_retry_delay:
                 delay = min(self.task.max_retry_delay, delay)
-        return self.end_date + delay
+        try:
+            return self.end_date + delay
+        except OverflowError:
+            return pendulum.DateTime.max
```

Comment:
Really feels like we are pushing the limits setting the next retry to year 9999... I wonder if we'd be better off building more realistic caps?

Reply:
What would be realistic? According to the exponential back-off in this case it's already off-limits. setting max is actually **before** the limit, so that things would not fail. Do you think we have another "realistic" date :) ?

Reply:
It maybe would have been better to have required max_retry_delay when exponential support was added, but that ship has sailed.

I have a hard time imagining more than a week is useful. Maybe the right answer is to have an instance wide default for this like `default_task_retry_delay`, with a default value of 1 week?

Reply:
I agree with Jed; the retry process is already at a dead end when OverflowError is encountered. Setting the date to 9999-12-31 fixes the immediate crash, but still means the retry would be stuck in the system forever (ish). It’d be better to either have a hard-limit on retry backoff (as proposed), or just give up retrying entirely (and mark the ti as failed) when backoff goes past a certain threshold.

Reply:
What about requiring [`max_retry_delay`](https://github.com/apache/airflow/blob/main/airflow/models/baseoperator.py#L464) if exponential backoff is set?